### ISSUE TITLE
File path infer problem

### ISSUE THREAD

**Issue by suziwen** [gatsbyjs/gatsby#13267]
Assuming I have three node

```
A = {
internal: {type: "File"},
children: [B],
parent: null,
absolutePath: '/home/test/a'
}

B = {
internal: {type: "File"},
children: [C],
parent: A
absolutePath: '/home/test/b'
}
C = {
internal: {type: "Custom"}
children: [],
parent: B,
cover: './featured.png'
}
```

I hope C's property (cover) will be inferred to `/home/test/b/featured.png`, but not  `/home/test/a/featured.png`.

Is it possible get the `/home/test/b/featured.png` cover?
or 
have any custom condition function to stop find root node?

**Comment by stefanprobst** [comment]
Could you provide a small reproduction repo to investigate? This would be very helpful, thanks!

**Comment by suziwen** [comment]
Reproduction link: https://github.com/suziwen/gatsby-file-path-infer-problem

-------------

```
{
  allCustomJson{
    edges{
      node{
        id
        parent{
          id
        }
        cover{
          absolutePath
        }
      }
    }
  }
  allFile{
    edges{
      node{
        id
        children{
          id
        }
        absolutePath
        parent{
          id
        }
      }
    }
  }
}
```

![image](https://user-images.githubusercontent.com/57370/55929393-05bdb000-5c4f-11e9-9e77-4f3265052997.png)

**Comment by stefanprobst** [comment]
Thanks for the reproduction, this should be fixed with #13289

**Comment by pieh** [comment]
Published `gatsby@2.3.19`
